models/login.go

```diff
@@ -242,17 +242,30 @@ func LoginUserLDAPSource(u *User, name, passwd string, source *LoginSource, auto
 	}
 
 	// Fallback.
+	if len(name) == 0 {
+		name = loginName
+	}
 	if len(mail) == 0 {
 		mail = fmt.Sprintf("%s@localhost", name)
 	}
+	var fullName string
```

Comment:
I think it would be better to put into a function like `composeFullName(fn,sn) string`